lib/openstack_query/runners/runner_wrapper.py

```diff
@@ -173,17 +205,21 @@ def _run_paginated_query(
         return query_res
 
     @staticmethod
-    def _apply_client_side_filter(
-        items: List[OpenstackResourceObj], filter_func: ClientSideFilterFunc
+    def _apply_client_side_filters(
+        items: List[OpenstackResourceObj], filters: ClientSideFilters
     ) -> List[OpenstackResourceObj]:
         """
-        Removes items from a list by running a given filter function
+        Removes items from a list by running a given filter functions
         :param items: List of items to query e.g. list of servers
-        :param filter_func: An Optional function that we can use to limit the results after querying openstacksdk,
-            - function takes an openstack resource object and returns True if it passes the filter, false if not
+        :param filters: filter functions that we can use to limit the results after querying openstacksdk,
+            - each function takes an openstack resource object and returns True if it passes the filter, false if not
         :return: List of items that match the1 given query
         """
-        return [item for item in items if filter_func(item)]
+        filtered_items = []
+        for item in items:
+            if all(filter_func(item) for filter_func in filters):
```

Comment:
Hmm, I don't like this, it took me three attempts to realise it's asking us the caller to filter despite being a filter function

Can we not get filter_func to do it for us:

```
def _apply(input_items: List):
  for filter in client_side_filters:
       input_items = filter_func(input_items)

# Where filter_func is implemented as
def filter_func(...):
    return filter(input, lambda)
```